packages/mcp/src/server/tools/icon-metadata-service.ts

```diff
@@ -0,0 +1,153 @@
+
+  static async getIconMetadata(): Promise<IconMetadata[]> {
+    if (this.cachedIconMetadata) {
+      console.error("[Vibe MCP] Using cached icon metadata");
+      return this.cachedIconMetadata;
+    }
+
+    try {
+      console.error("[Vibe MCP] Fetching icon metadata from unpkg...");
+      const metadata = await this.fetchWithRetry();
+      console.error(`[Vibe MCP] Fetched ${metadata.length} icons, caching for session`);
+      this.cachedIconMetadata = metadata;
+      return metadata;
+    } catch (error) {
+      console.error("[Vibe MCP] Failed to fetch icon metadata:", error);
+      console.error("[Vibe MCP] Trying to read local icon metadata...");
+
+      // Fallback to local metadata from the workspace
+      try {
+        const localMetadata = await this.getLocalIconMetadata();
+        this.cachedIconMetadata = localMetadata;
+        return localMetadata;
+      } catch (localError) {
+        console.error("[Vibe MCP] Failed to read local icon metadata:", localError);
+        console.error("[Vibe MCP] Returning empty array as fallback");
+        return [];
+      }
+    }
+  }
+
+  private static async fetchWithRetry(): Promise<IconMetadata[]> {
+    // Try fetch first
+    try {
+      const response = await fetch(this.UNPKG_URL);
+
+      if (!response.ok) {
+        throw new Error(`HTTP ${response.status}: ${response.statusText}`);
+      }
+
+      return await response.json();
+    } catch (fetchError) {
+      console.error("[Vibe MCP] Fetch failed (likely corporate proxy/SSL), trying curl fallback...");
+
+      // Fallback to curl for corporate networks
+      try {
+        const { stdout } = await execAsync(`curl -s -L "${this.UNPKG_URL}"`);
+        return JSON.parse(stdout);
+      } catch (curlError) {
+        console.error("[Vibe MCP] Curl fallback also failed:", curlError);
+        throw fetchError; // Re-throw original fetch error
+      }
+    }
+  }
+
+  private static async getLocalIconMetadata(): Promise<IconMetadata[]> {
+    try {
+      // Import the icons metadata directly if available
+      const path = await import("path");
+
+      // Get the workspace root (assuming we're in packages/mcp)
+      const workspaceRoot = path.resolve(process.cwd(), "../..");
+      const iconMetadataPath = path.join(workspaceRoot, "packages/icons/src/iconsMetaData.ts");
+
+      console.error(`[Vibe MCP] Attempting to import icon metadata from workspace...`);
+
+      // For now, return a basic set of common icons if we can't read the full metadata
+      // This provides a fallback when the full metadata isn't available
+      return this.getFallbackIconMetadata();
+    } catch (error) {
+      console.error("[Vibe MCP] Error reading local icon metadata:", error);
+      throw error;
+    }
+  }
+
+  private static getFallbackIconMetadata(): IconMetadata[] {
+    // Provide some common icons as fallback
+    return [
+      {
+        name: "Add",
+        file: "Add.svg",
```

Comment:
let's remove this
it's better to not give data instead of giving partial data
we also have a fallback curl so the chances for failing are not high